### PR TITLE
Add customdomain qa.register-trainee-teachers.education.gov.uk

### PR DESCRIPTION
**Pull request by felixtheflex**
### Context

Add custom domain for qa.register-trainee-teachers.education.gov.uk

### Changes proposed in this pull request

* Add custom domain for qa.register-trainee-teachers.education.gov.uk
* Create new route

### CODE REVIEW

File: terraform/modules/paas/data.tf [hunk 1]
```diff
@@ -18,3 +18,7 @@ data cloudfoundry_service postgres {
 data cloudfoundry_service redis {
   name = "redis"
 }
+
+data cloudfoundry_domain find_service_gov_uk {
```
Comment (vigneshmsft) [inline terraform/modules/paas/data.tf]:
```suggestion
data cloudfoundry_domain register_education_gov_uk {
```

File: terraform/modules/paas/data.tf [hunk 2]
```diff
@@ -18,3 +18,7 @@ data cloudfoundry_service postgres {
 data cloudfoundry_service redis {
   name = "redis"
 }
+
+data cloudfoundry_domain register_service_gov_uk {
```
Comment (vigneshmsft) [inline terraform/modules/paas/data.tf:22]:
```suggestion
data cloudfoundry_domain register_education_gov_uk {
```

File: terraform/modules/paas/variables.tf [hunk 3]
```diff
@@ -36,4 +36,12 @@ locals {
   worker_app_start_command = "bundle exec sidekiq -C config/sidekiq.yml"
   worker_app_name          = "register-worker-${var.app_environment}"
   logging_service_name     = "register-logit-${var.app_environment}"
+
+  service_gov_uk_host_names = {
+    qa      = ["qa"]
+    staging = ["staging"]
+    prod    = ["www", "www2"]
```
Comment (vigneshmsft) [inline terraform/modules/paas/variables.tf:43]: you don't need `www2` and also array of hostnames

File: terraform/modules/paas/main.tf [hunk 4]
```diff
@@ -72,6 +78,16 @@ resource cloudfoundry_route web_app_route {
   hostname = local.web_app_name
 }
 
+
+
+resource cloudfoundry_route web_app_service_gov_uk_route {
+  for_each = toset(lookup(local.service_gov_uk_host_names, var.app_environment))
```
Comment (vigneshmsft) [inline terraform/modules/paas/main.tf:84]: don't think you require multiple routes here

File: terraform/modules/paas/variables.tf (hunk 3 above)
Comment (felixtheflex) [reply terraform/modules/paas/variables.tf:43]: ok.. i thought that was needed by PaaS..

File: terraform/modules/paas/main.tf [hunk 5]
```diff
@@ -72,6 +78,16 @@ resource cloudfoundry_route web_app_route {
...
 }
 
+
+
+resource cloudfoundry_route web_app_service_gov_uk_route {
+  #for_each = toset(lookup(local.service_gov_uk_host_names, var.app_environment))
+
+  domain   = data.cloudfoundry_domain.register_education_gov_uk.id
+  space    = data.cloudfoundry_space.space.id
+  hostname = lookup(local.service_gov_uk_host_names, var.app_environment)
```
Comment (vigneshmsft) [inline terraform/modules/paas/main.tf:86]:
```suggestion
  hostname = local.service_gov_uk_host_names[var.app_environment]
```
also remove line 84

File: terraform/modules/paas/variables.tf [hunk 6]
```diff
@@ -36,4 +36,12 @@ locals {
...
   worker_app_name          = "register-worker-${var.app_environment}"
   logging_service_name     = "register-logit-${var.app_environment}"
+
+  service_gov_uk_host_names = {
+    qa      = "qa"
+    staging = "staging"
+    prod    = "www"
+  }
+  web_app_service_gov_uk_route_ids = [cloudfoundry_route.web_app_service_gov_uk_route.id]
+  web_app_routes                   = concat(local.web_app_service_gov_uk_route_ids, [cloudfoundry_route.web_app_route.id])
```
Comment (vigneshmsft) [inline terraform/modules/paas/variables.tf:46]:
```suggestion
  web_app_routes                   = [cloudfoundry_route.web_app_service_gov_uk_route.id, cloudfoundry_route.web_app_route.id]
```

**Review by felixtheflex — COMMENTED** [review]
foo

File: terraform/modules/paas/main.tf (hunk 5 above)
Comment (felixtheflex) [reply terraform/modules/paas/main.tf]: resolved

**Review by defong — APPROVED** [review]

**Review by vigneshmsft — APPROVED** [review]
LG

**Pull request merged**